src/components/GeolocationTracking/GeolocationTrackingSwitcher.jsx

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable no-console */
+import React, { useState, useEffect } from 'react'
+
+import { useClient } from 'cozy-client'
+import ConnectionFlow from 'cozy-harvest-lib/dist/models/ConnectionFlow'
+import { useWebviewIntent } from 'cozy-intent'
+import FormControlLabel from 'cozy-ui/transpiled/react/FormControlLabel'
+import Switch from 'cozy-ui/transpiled/react/Switch'
+import useBreakpoints from 'cozy-ui/transpiled/react/hooks/useBreakpoints'
+import { makeStyles } from 'cozy-ui/transpiled/react/styles'
+
+const KONNECTOR_DOCTYPE = 'io.cozy.konnectors'
+
+const generateId = () => Math.random().toString(36).replace('0.', '')
+
+const useStyles = makeStyles(() => ({
+  root: {
+    marginLeft: 0
+  },
+  labelPlacementStart: {
+    display: 'flex',
+    justifyContent: 'space-between'
+  }
+}))
+
+export const GeolocationTrackingSwitcher = ({ className }) => {
+  const webviewIntent = useWebviewIntent()
+  const client = useClient()
+  const { isMobile } = useBreakpoints()
+  const classes = useStyles()
+
+  const [isGeolocationTrackingEnabled, setIsGeolocationTrackingEnabled] =
+    useState(false)
+
+  const handleGeolocationTrackingChange = async () => {
+    if (isGeolocationTrackingEnabled) {
+      // disable geolocation tracking
+      await webviewIntent.call('setGeolocationTracking', false)
+    } else {
+      // create account if necessary
+      let geolocationTrackingId = await getGeolocationTrackingId()
+
+      if (geolocationTrackingId === null) {
+        const { deviceName } = await webviewIntent.call('getDeviceInfo')
+
+        const {
+          data: { attributes: konnector }
+        } = await client.collection(KONNECTOR_DOCTYPE).get('tracemob')
```

Comment:
client.query(Q('io.cozy.konnectors').getById('tracemob')) 

No direct call to `collection` 🙏